pkg/utils/status.go

```diff
@@ -654,77 +688,246 @@ func WaitForNginxController(ctx context.Context, instance csmv1.ContainerStorage
+
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodRunning {
+			readyPods++
 		}
 	}
-	log.Infof("calculate overall state [%s] of module", newStatus.State)
 
-	return running, err
+	if readyPods == expected {
+		daemonRunning = true
+	}
+
+	if(certEnabled && veleroEnabled) {
+		return appMobRunning && certManagerRunning && certManagerCainInjectorRunning && certManagerWebhookRunning && veleroRunning && daemonRunning, nil
+	}
+
+	if(!certEnabled && !veleroEnabled) {
+		return appMobRunning && daemonRunning, nil
+	}
+
+	if(!certEnabled && veleroEnabled) {
+		return appMobRunning && daemonRunning && veleroRunning, nil
+	}
+
+	if(certEnabled && !veleroEnabled) {
+		return appMobRunning && certManagerCainInjectorRunning && certManagerRunning && certManagerWebhookRunning && daemonRunning, nil
+	}
+
+
+	return false, nil
+
+
+
 
 }
 
-// statusForAppMob - calculate success state for application-mobility module
-func statusForAppMob(ctx context.Context, instance *csmv1.ContainerStorageModule, r ReconcileCSM, newStatus *csmv1.ContainerStorageModuleStatus) (bool, error) {
+// observabilityStatusCheck - calculate success state for observability module
+func observabilityStatusCheck(ctx context.Context, instance *csmv1.ContainerStorageModule, r ReconcileCSM, newStatus *csmv1.ContainerStorageModuleStatus) (bool, error) {
+	//log := logger.GetLogger(ctx)
+	// Observability launches three pods in the karavi namespace
+	//expectedObservabilityPods := 3
+	//readyPods := 0
+	topologyEnabled := false
+	otelEnabled := false
+	certEnabled := false
+	metricsEnabled := false
+	certManagerRunning := false
+	certManagerCainInjectorRunning := false
+	certManagerWebhookRunning := false
+	otelRunning := false
+	metricsRunning := false
+	topologyRunning := false
 
-	running := false
-	var appRunning bool
-	var velRunning bool
-	var certRunning bool
-	var restRunning bool
-	var err error = nil
 	for _, m := range instance.Spec.Modules {
-		if m.Name == csmv1.ApplicationMobility {
-			appRunning, err = checkForServices(ctx, instance, r, newStatus, false)
+		if m.Name == csmv1.Observability {
 			for _, c := range m.Components {
-				if c.Name == "velero" {
+				if c.Name == "topology" {
 					if *c.Enabled {
-						velRunning, err = checkForServices(ctx, instance, r, newStatus, false)
-						if c.DeployNodeAgent {
-							restRunning, err = checkForServices(ctx, instance, r, newStatus, true)
-						}
+						topologyEnabled = true
 					}
-				} else if c.Name == "cert-manager" {
+				}
+				if c.Name == "otel-collector" {
 					if *c.Enabled {
-						certRunning, err = checkForServices(ctx, instance, r, newStatus, false)
+						otelEnabled = true
+					}
+				}
+				if c.Name == "cert-manager" {
+					if *c.Enabled {
+						certEnabled = true
+					}
+				}
+				if c.Name == fmt.Sprintf("metrics-%s",instance.Spec.Driver.CSIDriverType)  {
+					if *c.Enabled {
+						metricsEnabled = true
 					}
-
 				}
 			}
 		}
+	}
+
+	namespace := "karavi"
```

Comment:
Can we be sure that Observability will always be in this namespace?

Reply:
yes 
Thats what the document and observability config file indicates